### PR TITLE
Implement XML register packets for lldb

### PR DESCRIPTION
**Pull request by fjricci**

### CODE REVIEW

File: Sources/GDBRemote/DebugSessionImpl.cpp [hunk 1]
```diff
@@ -440,24 +438,47 @@ DebugSessionImpl::onQuerySharedLibrariesInfoAddress(Session &,
...
 }
 
-ErrorCode DebugSessionImpl::onXferRead(Session &, std::string const &object,
+ErrorCode DebugSessionImpl::onXferRead(Session &session,
+                                       std::string const &object,
                                        std::string const &annex,
                                        uint64_t offset, uint64_t length,
                                        std::string &buffer, bool &last) {
-  DS2LOG(Info, "object='%s' annex='%s' offset=%#llx length=%#llx",
+  DS2LOG(Warning, "object='%s' annex='%s' offset=%#llx length=%#llx",
```
Comment (sas) [inline Sources/GDBRemote/DebugSessionImpl.cpp]: Not sure this should stay a warning.

Comment (fjricci) [inline Sources/GDBRemote/DebugSessionImpl.cpp]: Oops yeah, left that in from debugging

**Comment by sas** [comment]
Looks like this broke the llgs tests.

**Comment by strager** [comment]
lgtm but I don't think I'm qualified to review the third commit.

**Comment by sas** [comment]
LGTM, will merge.

**Pull request merged**